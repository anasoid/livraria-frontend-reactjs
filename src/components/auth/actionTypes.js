export const AUTHENTICATE = 'AUTHENTICATE';
export const AUTH_ERROR = 'AUTH_ERROR';
export const SIGNOUT = 'SIGNOUT';