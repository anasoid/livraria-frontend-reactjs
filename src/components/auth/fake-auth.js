
export const fakeAuth = {
  authenticate(username, password) {
    const auth = username === 'admin@admin' && password === '123';
    return new Promise((resolve, reject) => setTimeout(auth?resolve(true):reject(false), 500)); // fake async
  },
  signout() {
    return new Promise(resolve => setTimeout(resolve(true), 500)); // fake async
  }
}