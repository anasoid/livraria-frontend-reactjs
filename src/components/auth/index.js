
export const fakeAuth = {
  isAuthenticated: false,
  username: '',
  authenticate(callBack, username) {
    this.isAuthenticated = true;
    this.username = username;
    setTimeout(callBack, 100); // fake async
  },
  signout(callBack) {
    this.isAuthenticated = false;
    setTimeout(callBack, 100); // fake async
  }
}