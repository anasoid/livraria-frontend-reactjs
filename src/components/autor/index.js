import React from 'react';
import AutorContainer from './autor-container';

const Autor = () => {
  return (
    <AutorContainer/>
  );
};

export default Autor;