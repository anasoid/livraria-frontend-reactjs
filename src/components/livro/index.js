import React from 'react';
import LivroContainer from './livro-container';

const Livro = () => {
  return (
    <LivroContainer/>
  );
}

export default Livro;