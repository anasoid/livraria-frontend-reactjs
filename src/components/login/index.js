import React from 'react';
import LoginContainer from './login-container';

const Login = () => {
  return (
    <LoginContainer/>
  );
}

export default Login;