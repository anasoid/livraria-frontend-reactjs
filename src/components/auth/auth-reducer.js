import {AUTH_ERROR, AUTHENTICATE, SIGNOUT} from './actionTypes';

const initialState = {
  isAuthenticated: false,
  authError: '',
  username: ''
};

const authReducer = (state = initialState, action) => {
  switch (action.type) {
    case AUTHENTICATE:
      return {
        ...state,
        isAuthenticated: true,
        username: action.payload.username
      };
    case AUTH_ERROR:
      return {
        ...state,
        isAuthenticated: false,
        authError: 'Usuario ou Senha incorretos'
      };
    case SIGNOUT:
      return initialState;
    default:
      return state;
  }
};

export default authReducer;