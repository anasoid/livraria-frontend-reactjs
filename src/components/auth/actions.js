import {AUTHENTICATE, AUTH_ERROR, SIGNOUT} from './actionTypes';
import {fakeAuth} from './fake-auth';

export const authenticate = (username, password) => {
  return dispatch => {
    return fakeAuth.authenticate(username, password)
      .then(() => {
        return dispatch({type: AUTHENTICATE, payload: { username, password }});
      }).catch( () => dispatch({type: AUTH_ERROR}));
  };
};

export const logout = () => {
  return dispatch => {
    return fakeAuth.signout()
      .then(() => {
        return dispatch({type: SIGNOUT});
      }).catch();
  };
};