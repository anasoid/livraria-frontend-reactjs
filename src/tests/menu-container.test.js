import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import MenuContainer, {_handleLogout} from '../livraria-header/menu-container';
import { MemoryRouter } from 'react-router';
import {fakeAuth} from '../components/auth';


describe('test MenuContainer component', () => {

  const wrapper = shallow(<MenuContainer/>);

  it('should renders without crashing ', () => {
    shallow(<MenuContainer/>);
    const tree = renderer.create(<MemoryRouter><MenuContainer/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be menu item Livro', () => {
    const livro = wrapper.find('#livroItem');
    expect(livro).toMatchSelector('Link');
    expect(livro).toHaveProp('children', 'Livros');
    expect(livro).toHaveProp('to', '/livros');
  });

  it('should be menu item Autores', () => {
    const autor = wrapper.find('#autorItem');
    expect(autor).toMatchSelector('Link');
    expect(autor).toHaveProp('children', 'Autores');
    expect(autor).toHaveProp('to', '/autores');
  });

  it('should be menu item Vendas', () => {
    const venda = wrapper.find('#vendaItem');
    expect(venda).toMatchSelector('Link');
    expect(venda).toHaveProp('children', 'Vendas');
    expect(venda).toHaveProp('to', '/vendas');
  });

  it('should be menu item Logout', () => {
    const logout = wrapper.find('#logoutItem');
    expect(logout).toMatchSelector('Link');
    expect(logout).toHaveProp('children', 'Logout');
    expect(logout).toHaveProp('to', '/');
  });

});

describe('test logout menu', () => {

  it('should call fakeauth.signout when logout', () => {

    //given
    const signoutAuthMock = jest.fn();
    fakeAuth.signout = signoutAuthMock;

    //when
    _handleLogout();

    //then
    expect(signoutAuthMock).toBeCalled();

  });

});