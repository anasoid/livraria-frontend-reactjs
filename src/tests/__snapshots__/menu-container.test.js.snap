// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MenuContainer component should renders without crashing  1`] = `
<nav
  className="menu"
>
  <ul>
    <li>
      <a
        href="/livros"
        id="livroItem"
        onClick={[Function]}
      >
        Livros
      </a>
    </li>
    <li>
      <a
        href="/autores"
        id="autorItem"
        onClick={[Function]}
      >
        Autores
      </a>
    </li>
    <li>
      <a
        href="/vendas"
        id="vendaItem"
        onClick={[Function]}
      >
        Vendas
      </a>
    </li>
    <li>
      <a
        href="/"
        id="logoutItem"
        onClick={[Function]}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;
